clientv3/client.go

```diff
@@ -519,13 +519,17 @@ func (c *Client) roundRobinQuorumBackoff(waitBetween time.Duration, jitterFracti
 
 func (c *Client) checkVersion() (err error) {
 	var wg sync.WaitGroup
-	errc := make(chan error, len(c.cfg.Endpoints))
+
+	eps := c.Endpoints()
+	errc := make(chan error, len(eps))
 	ctx, cancel := context.WithCancel(c.ctx)
 	if c.cfg.DialTimeout > 0 {
-		ctx, cancel = context.WithTimeout(ctx, c.cfg.DialTimeout)
+		cancel()
```

Comment:
I believe the behavior is the same with and without this part of the change?

Reply:
Yes, it doesn't change any behavior, I believe. Just trying to be clear about what we are doing, overwriting the previous cancel function.

Reply:
Agree that we can go with or without cancel() here.